examples/solvers/proximal_gradient_wavelet_tomography.py

```diff
@@ -0,0 +1,118 @@
+space = odl.uniform_discr(
+    min_pt=[-20, -20], max_pt=[20, 20], shape=[256, 256], dtype='float32')
+
+# Make a parallel beam geometry with flat detector
+# Angles: uniformly spaced, n = 300, min = 0, max = 2 * pi
+angle_partition = odl.uniform_partition(0, 2 * np.pi, 300)
+# Detector: uniformly sampled, n = 300, min = -30, max = 30
+detector_partition = odl.uniform_partition(-30, 30, 300)
+geometry = odl.tomo.Parallel2dGeometry(angle_partition, detector_partition)
+
+# The implementation of the ray transform to use, options:
+# 'scikit'                    Requires scikit-image (can be installed by
+#                             running ``pip install scikit-image``).
+# 'astra_cpu', 'astra_cuda'   Requires astra tomography to be installed.
+#                             Astra is much faster than scikit. Webpage:
+#                             https://github.com/astra-toolbox/astra-toolbox
+impl = 'astra_cuda'
+
+# Create the forward operator, and also the vectorial forward operator.
+ray_trafo = odl.tomo.RayTransform(space, geometry, impl=impl)
+
+
+# --- Generate artificial data --- #
+
+
+# Create phantom
+discr_phantom = odl.phantom.shepp_logan(space, modified=True)
+
+# Create sinogram of forward projected phantom with noise
+data = ray_trafo(discr_phantom)
+data += odl.phantom.white_noise(ray_trafo.range) * np.mean(data) * 0.1
+
+
+# --- Set up the inverse problem --- #
+
+
+# Create wavelet operator
+W = odl.trafos.WaveletTransform(space, wavelet='haar', nlevels=5)
+
+# The wavelets bases are normalized to constant norm regardless of scale.
+# since we want to penalize "small" wavelets more than "large" ones, we need
+# to weight by the scale of the wavelets.
+# The "area" of the wavelets scales as 2 ^ scale, but we use a slightly smaller
+# number in order to allow some high frequencies.
+scales = W.inverse.get_scales()
+Wtrafo = W.inverse * (1 / (np.power(1.7, 1 + scales)))
+
+# Create regularizer as l1 norm
+regularizer = 0.0002 * odl.solvers.L1Norm(W.range)
+
+# l2-squared norm of residual
+l2_norm_sq = odl.solvers.L2NormSquared(ray_trafo.range).translated(data)
+
+# Compose from the right with ray transform and wavelet transform
+data_discrepancy = l2_norm_sq * ray_trafo * Wtrafo
+
+# --- Select solver parameters and solve using proximal gradient --- #
+
+# Select step-size that guarantees convergence.
+gamma = 0.2
+
+# Optionally pass callback to the solver to display intermediate results
+callback = (odl.solvers.CallbackPrintIteration() &
+            odl.solvers.CallbackShow(display_step=10))
+
+
+def callb(x):
+    """Callback that displays the inverse wavelet transform of current iter."""
+    callback(Wtrafo(x))
+
+# Run the algorithm (FISTA)
+x = data_discrepancy.domain.zero()
+odl.solvers.accelerated_proximal_gradient(
+    x, f=regularizer, g=data_discrepancy, niter=200, gamma=gamma,
+    callback=callb)
+
+# Display images
+data.show(title='Data')
+x.show(title='Wavelet coefficients')
```

Comment:
Hm, interesting suggestion. How would you go about it? Using `x`, `y`, `z` for up to 3d and `[x{}.format(i + 1) for i in range(self.ndim)]` for higher dimensions? The show implementation already selects the correct labels depending on the way you slice, so that wouldn't require any new code.